onnxruntime/core/optimizer/selectors_actions/helpers.h

```diff
@@ -11,14 +11,52 @@ namespace onnxruntime {
 
 // Struct to serialize the node indexes in an ORT format model.
 // Use EmptyNodeIndex for nullptr entries in the vectors for missing optional inputs
-struct NodesToOptimizeIndexes {
-  std::vector<NodeIndex> nodes;
-  int num_inputs;
-  int num_outputs;
-  bool variadic_input;
-  bool variadic_output;
-  int num_variadic_inputs;
-  int num_variadic_outputs;
+class NodesToOptimizeIndexes {
+ public:
+  const std::vector<NodeIndex>& GetNodes() const { return nodes_; }
```

Comment:
>const [](http://example.com/codeflow?start=43&length=5)

another idea is to make all of the NodesToOptimizeIndexes struct's data members const - wouldn't need these accessors then

Reply:
Moved to struct, probably will move back to accessors if later we need to add more helpers to NodesToOptimizeIndexes (similar to NodesToOptimize)